### PR TITLE
Sched 150: App structure

### PR DESCRIPTION
**Pull request by stroncod**

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import signal
+from datetime import datetime
+from .process_manager import ProcessManager, TaskType
+from .scheduler import Scheduler
+
+TASK_QUEUE = []
+
+
+class App:
+    def __init__(self, config):
```
Comment (sraaphorst) [inline app/app.py:11]: Can we say anything about the type of `config`?

File: app/config.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass
+from astropy.time import Time
+from astropy.units import Quantity
+
+
+@dataclass
+class CollectorConfig:
+    semesters: set
+    program_types: set
+    obs_classes: set
```
Comment (sraaphorst) [inline app/config.py:10]: We should make these `FrozenSet[Semester]`, etc.

File: app/config.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+@dataclass
+class SelectorConfig:
+    properties: type
```
Comment (sraaphorst) [inline app/config.py:15]: Can we say anything more than `type`?

File: app/config.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+@dataclass
+class SchedulerConfig:
+    start_time: Time
+    end_time: Time
+    time_slot_length: Quantity
+    sites: set
```
Comment (sraaphorst) [inline app/config.py:23]: `FrozenSet[Site]`

**Review by sraaphorst — COMMENTED** [review]
This looks good to me, except for concerns about the type hints.
Did you leave them out for a reason?

**Review by sraaphorst — APPROVED** [review]

File: app/app.py (hunk 1 above)
Comment (stroncod) [reply app/app.py:11]: This is address in #120

File: app/config.py (hunk 2 above)
Comment (stroncod) [reply app/config.py:10]: This is removed in #120

File: app/config.py (hunk 3 above)
Comment (stroncod) [reply app/config.py:15]: Same as above

**Pull request merged**